packages/client/tests/functional/extensions/tx.ts

```diff
@@ -191,7 +191,21 @@ testMatrix.setupTestSuite((_0, _1, clientMeta) => {
         isolationLevel: 'Serializable',
       })
 
-      expectTypeOf(data).toEqualTypeOf<({ id: string } | null)[]>()
+      expectTypeOf(data).toEqualTypeOf<[{ id: string } | null]>()
+    }
+  })
+
+  test('type inference allows for destructuring the array', () => {
+    ;async () => {
```

Comment:
nit: I don't understand why this test needs `;async () => {` like this. We usually do it like this
```
test('description', async () => {
```

Reply:
Yep, so this is because we don't really care to run it when it's type-level only

Reply:
But it is still convenient to write it as a jest test even if it does not matter